### PR TITLE
Start on making LoFTR jit script compatible.

### PR DESCRIPTION
**Pull request by more-pixels-please**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/feature/loftr/loftr.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def __init__(self, pretrained: Optional[str] = 'outdoor', config: Dict = default
             self.load_state_dict(pretrained_dict['state_dict'])
         self.eval()
 
-    def forward(self, data: dict) -> Dict[str, torch.Tensor]:
+    def forward(self, data: Dict[str, torch.Tensor]):
```
Comment (edgarriba) [inline kornia/feature/loftr/loftr.py:103]:
```suggestion
    def forward(self, data: Dict[str, Tensor]):
```
you can do now as follows `from kornia.core import Tensor`

we are isolating torch imports to have a better understanding what we consume from pytorch + adding code cleaning

File: kornia/feature/loftr/loftr.py [hunk 2]
```diff
@@ -119,20 +119,20 @@ def forward(self, data: dict) -> Dict[str, torch.Tensor]:
         """
 
         # 1. Local Feature CNN
-        data.update({
-            'bs': data['image0'].size(0),
-            'hw0_i': data['image0'].shape[2:], 'hw1_i': data['image1'].shape[2:]
-        })
+        bs = torch.tensor(data['image0'].size(0))
```
Comment (edgarriba) [inline kornia/feature/loftr/loftr.py:122]:
```suggestion
        bs = torch.as_tensor(data['image0'].size(0))
```
to avoid copy

Comment (edgarriba) [reply kornia/feature/loftr/loftr.py:122]: are shapes really required to be converted to tensors ? maybe the dict can support `Any` ? /cc @ducha-aiki

**Review by edgarriba — COMMENTED** [review]
how are you actually testing this ? I'd suggest to uncomment the following line
https://github.com/kornia/kornia/blob/master/test/feature/test_loftr.py#L69

one more suggestion is to add typing as much as possible in the function signatures, and then in the code where you feel it will help for readability etc, but not over typed.

File: kornia/feature/loftr/loftr.py (hunk 2 above)
Comment (factify) [reply kornia/feature/loftr/loftr.py:122]: i believe dict can support `Any` but then you end up possibly duplicating code as you'd have to run a condition for jit scripting like so:
`if torch.jit.isinstance(x, Dict[str, torch.Tensor]):`
not sure if there is a workaround.

File: kornia/feature/loftr/loftr.py (hunk 1 above)
Comment (factify) [reply kornia/feature/loftr/loftr.py:103]: thanks, i'll change that.
for testing, i'll use your suggestion.

**Comment by factify** [comment]
Running 'TestLoFTR.test_jit()' will result in the following:
`RuntimeError: Unsupported value kind: Tensor`

**Comment by factify** [comment]
@edgarriba @ducha-aiki, when i run jit.trace i get the following:
`RuntimeError: 0INTERNAL ASSERT FAILED at "C:\\actions-runner\\_work\\pytorch\\pytorch\\builder\\windows\\pytorch\\torch\\csrc\\jit\\ir\\alias_analysis.cpp":607, please report a bug to PyTorch. We don't have an op for aten::fill_ but it isn't a special case.  Argument types: Tensor, bool,

Candidates:
        aten::fill_.Scalar(Tensor(a!) self, Scalar value) -> (Tensor(a!))
        aten::fill_.Tensor(Tensor(a!) self, Tensor value) -> (Tensor(a!))`
not sure where this actually occurs, but according to some formus (https://github.com/pytorch/pytorch/issues/67834), Tensors of bool types cause this error. is this what you observed?

**Comment by edgarriba** [comment]
never experienced that -- i can try to run it during the weekend

**Comment by factify** [comment]
So far the following modules are jit scriptable; _backbone, LocalFeatureTransformer_ and _FinePreprocess_. The _PositionEncodingSine_ module has got a "register_buffer" call which complicates things. The _Coarse/FineMatching_ modules both have dict "updates" which are also not scriptable.

**Comment by edgarriba** [comment]
@factify possibly next step is to get rid of dicts in a reasonable way

**Comment by edgarriba** [comment]
just for the record here, worth to start planning to test this with torch dynamo. torchscrtipt is gonna be deprecated soon 
https://github.com/kornia/kornia/pull/2013

**Comment by xmba15** [comment]
@edgarriba @factify any new updates on this task?

**Comment by johnnv1** [comment]
> @edgarriba @factify any new updates on this task?

we are dropping JIT support on kornia in favor of dynamo, so, I think this patch will not go ahead (#2200)

**Comment by edgarriba** [comment]
> we are dropping JIT support on kornia in favor of dynamo, so, I think this patch will not go ahead (#2200)

worth to check though that torchdynamo can compile this module

**Pull request closed without merging**